packages/typescript/src/watchProgram.ts

```diff
@@ -162,6 +162,10 @@ function createWatchHost(
     ...baseHost,
     /** Override the created program so an in-memory emit is used */
     afterProgramCreate(program) {
+      // Ensure we recompute custom transformers for each new builder program in watch mode
+      // so factories capture the current Program/TypeChecker and any provided getters return
+      // the latest values. This avoids freezing the initial Program across rebuilds.
+      createdTransformers = undefined;
```

Comment:
@CharlieHelps the linting rules for this repo count use of `undefined` as a warning. please use `void 0` in its place. please check your other work for similar warnings. don't modify code that existed before your changes in this PR, however.